### PR TITLE
Service broker preserves policy by default on deprovision

### PR DESCRIPTION
**Pull request by micahlee**
Connected to #97 

This PR removes the broker configuration option for `CONJUR_PRESERVE_POLICY` and will always preserve the policy on deprovision. This means the service broker will load new policy during provisioning to define policy branches and layers for the Org and Space, but the service broker will never attempt to delete them.

### CODE REVIEW

File: features/integration.feature [hunk 1]
```diff
@@ -8,7 +8,6 @@ Feature: Integration Tests for PCF 2.4
 
   Scenario: Service broker functions correctly with PCF 2.4 
     When I create a service instance for Conjur
-    Then the instance resource exists
     And the policy for the org and space exists
```
Comment (izgeri) [inline features/integration.feature]: And => Then?

File: features/integration.feature [hunk 2]
```diff
@@ -20,24 +19,11 @@ Feature: Integration Tests for PCF 2.4
...
-
     When I remove the service instance
     Then the policy for the org and space exists
-    And the instance resource doesn't exist
 
-  @preserve-policy
   Scenario: Redeploying service broker
     When I create a service instance for Conjur
-    Then the instance resource exists
     And the policy for the org and space exists
```
Comment (izgeri) [inline features/integration.feature]: And => Then

File: spec/controllers/provision_controller_spec.rb [hunk 3]
```diff
@@ -81,94 +78,4 @@
       end
     end
   end
-
-  describe 'DELETE' do
```
Comment (izgeri) [inline spec/controllers/provision_controller_spec.rb]: do we want to have any tests in for when we receive a deprovision request?

Comment (micahlee) [reply spec/controllers/provision_controller_spec.rb]: The only test case we have now that would remain is `returns with a 200 OK response`, and this is also verified in the cuke tests here: https://github.com/cyberark/conjur-service-broker/blob/master/features/deprovision.feature#L11.

I don't hold strongly one way or the other though, so I'm happy to leave that spec in.

**Comment by micahlee** [comment]
Thanks @izgeri for the comments! I've updated the PR to incorporate the updates. FYI, I won't have a clean build in Jenkins with the integration test until we finish or pause the XA testing using the PCF cluster.

**Pull request merged**